crates/top/rerun/src/commands/rrd/stats.rs

```diff
@@ -0,0 +1,397 @@
+use itertools::Itertools as _;
+
+use re_protos::log_msg::v1alpha1::log_msg::Msg;
+use re_viewer::external::egui::ahash::{HashMap, HashMapExt as _};
+
+use crate::commands::read_raw_rrd_streams_from_file_or_stdin;
+
+// ---
+
+#[derive(Debug, Clone, clap::Parser)]
+pub struct StatsCommand {
+    /// Paths to read from. Reads from standard input if none are specified.
+    path_to_input_rrds: Vec<String>,
+
+    /// If set, will try to proceed even in the face of IO and/or decoding errors in the input data.
+    #[clap(long = "continue-on-error", default_value_t = true)]
+    continue_on_error: bool,
+}
+
+impl StatsCommand {
+    pub fn run(&self) -> anyhow::Result<()> {
+        let Self {
+            path_to_input_rrds,
+            continue_on_error,
+        } = self;
+
+        let mut num_chunks = 0u64;
+        let mut num_chunks_per_entity: HashMap<String, u64> = HashMap::new();
+        let mut num_chunks_per_index: HashMap<String, u64> = HashMap::new();
+        let mut num_chunks_per_component: HashMap<String, u64> = HashMap::new();
+        let mut num_rows = Vec::with_capacity(num_chunks as _);
+        let mut num_static = 0u64;
+        let mut num_indexes = Vec::with_capacity(num_chunks as _);
+        let mut num_components = Vec::with_capacity(num_chunks as _);
+        let mut ipc_size_bytes_compressed = Vec::with_capacity(num_chunks as _);
+        let mut ipc_size_bytes_uncompressed = Vec::with_capacity(num_chunks as _);
+        let mut ipc_schema_size_bytes_uncompressed = Vec::with_capacity(num_chunks as _);
+        let mut ipc_data_size_bytes_uncompressed = Vec::with_capacity(num_chunks as _);
+
+        let (rx, _) = read_raw_rrd_streams_from_file_or_stdin(path_to_input_rrds);
+
+        re_log::info!("processing input…");
+        let mut last_checkpoint = std::time::Instant::now();
+        for (_source, res) in rx {
+            let mut is_success = true;
+
+            match res {
+                Ok(msg) => {
+                    num_chunks += 1;
```

Comment:
that's technically arbitrary messages an not chunks as not every message is a chunk. Not that it makes much difference in practice. Shouldn't that go only in the  Ok(Some()) below? Since that's only used for rate printing it would be more accurate to just call it `num_msg_processed` and print as much